File: 504. Base 7/C++/Solution.cpp
class Solution {
public:
    string convertToBase7(int num) {
        if (num == 0) {
            return "0";
        }
        int sign = num >= 0 ? 1 : -1;
        num = abs(num);
        string result = "";
        while (num != 0) {
            result = to_string(num % 7) + result;
            num /= 7;
        }

        return (sign < 0 ? "-" : "") + result;
    }
};

File: 504. Base 7/Java/Solution.java
public class Solution {
    public String convertToBase7(int num) {
        if (num == 0) {
            return "0";
        }
        int sign = num >= 0 ? 1 : -1;
        num = Math.abs(num);
        StringBuilder builder = new StringBuilder();
        while (num != 0) {
            builder.append(num % 7);
            num /= 7;
        }
        if (sign < 0) {
            builder.append('-');
        }
        return builder.reverse().toString();
    }
}